src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -197,7 +202,33 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
 
                     while (await reader.ReadAsync(cancellationToken))
                     {
-                        (short resourceTypeId, string resourceId, int version, bool isDeleted, long resourceSurrogateId, string requestMethod, bool isMatch, bool isPartialEntry, Stream rawResourceStream) = reader.ReadRow(
+                        short resourceTypeId;
+                        string resourceId;
+                        int version;
+                        bool isDeleted;
+                        long resourceSurrogateId;
+                        string requestMethod;
+                        bool isMatch;
+                        bool isPartialEntry;
+                        bool rawResourceMetaSet = false;
+                        Stream rawResourceStream;
+
+                        if (_schemaInformation.Current <= 3)
```

Comment:
Another approach is to always add RawResourceMetaSet to the projection, and to make that false when version is less than 4.

Reply:
I'm not sure I understand, I'll sync up with you tomorrow about this.